File: java/valid_palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        int start = 0;
        int end = s.length()-1;
        
        if(s.length() <= 1){
            return true;
        }
        
        while(start < end){
            Character pt1 = s.charAt(start);
            Character pt2 = s.charAt(end);
            
            if(!Character.isLetterOrDigit(pt1)){
                start++;
                continue;
            }
            if(!Character.isLetterOrDigit(pt2)){
                end--;
                continue;
            }
            if(Character.toLowerCase(pt1) != Character.toLowerCase(pt2)){
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
}

File: cpp/valid_palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        vector<char> str;
        
        for(int i = 0; i < s.length(); ++i){
            if(isalnum(s[i])){
                str.push_back(tolower(s[i]));
            }
        }        
        
        for(int i = 0; i < str.size()/2; ++i){
            if(str[i] != str[str.size()-i-1]){
                return false;
            }
        }
        return true;
    }
};